int16[] start
int16[] end
